compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -460,55 +460,71 @@ static LogicalResult setRootConfigForPackPeel4LevelTilingPipeline(
     return linalgOp.emitOpError("failed to fetch m/n/k dims.");
   }
 
+  // Set attribute for number of input loops.
```

Comment:
I think I'm missing something here. You get the number of loops from linalgOp, and then set an attribute on the linalgOp. Why not just always call linalgOp.getNumLoops(); when you need it? 

Is it because  linalgOp.getNumLoops() changes as the op gets transformed, and you need to know what the original number of loops? If so the name should reflect that.

Reply:
No, it's because we need this in Passes.cpp, but we can't directly access linalgOp. Ref https://github.com/nod-ai/iree-amd-aie/pull/1098/files#diff-42f0d0bb098689f25ee68e8f05ec6c2eaa89ce41a6394d7d99c1d1c912943b38R313.

Reply:
As suggested above, I can bring a new pipeline if adding `num_loops` is hard to understand and debug,

Reply:
I see, thanks. 

I'm not really in favour of a new pipeline, I think it's not worth the duplication of code. Maybe it's possible to use the C++ language features to make this easy enough to understand (i.e. alternatives to if-else)? 

One option: instead of havinh the attribute num_loops, pass the `rootOp` into ` addPackPeel4LevelTilingBasedPassPipeline` directly? More direct and easier to understand IMO

Also... if we could have an identity linalg.pack there wouldn't even be any branching? Shouldn't we just request this change upstream?

Reply:
> One option: instead of havinh the attribute num_loops, pass the `rootOp` into ` addPackPeel4LevelTilingBasedPassPipeline` directly? More direct and easier to understand IMO

Yeah, I think this could work. I'll do a revision with that.

> Also... if we could have an identity linalg.pack there wouldn't even be any branching? Shouldn't we just request this change upstream?

We could discuss other options in the sync meeting.